docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+    ```
+
+4. Run the script:
+
+    ```bash
+    ./run.sh
+    ```
+
+5. The script will execute ANTA tests for each device in parallel, generating a JSON report for each device in the `nrfu_reports` directory.
+
+#### Tag-Based Implementation
+
+1. Create the tag-specific test runner (`nrfu-tag.sh`):
+
+    ```bash
+    #!/bin/bash
+    set -euo pipefail
+
+    exec 2>&1
+
+    usage() {
+        echo "Usage: $0 <tag-name> [<nrfu-reports-path>]"
+        exit 1
+    }
+
+    ANTA_INVENTORY="anta_inventory.yml"
+    ANTA_CATALOG="anta_catalog.yml"
+    TAG_NAME=${1:-}
+    NRFU_REPORT_PATH=${2:-nrfu_reports}
+
+    if [[ -z "$TAG_NAME" ]]; then
+        usage
+    fi
+
+    anta nrfu \
+        --tags "$TAG_NAME" \
+        -i "$ANTA_INVENTORY" \
+        -c "$ANTA_CATALOG" \
+        json -o "$NRFU_REPORT_PATH/${TAG_NAME}.json"
+    ```
+
+    Same as before, adjust `ANTA_INVENTORY`, `ANTA_CATALOG` to match your environment.
+
+2. Create the parallel execution script (`run-tag.sh`):
+
+    ```bash
+    #!/bin/bash
+
+    mkdir -p nrfu_reports
+
+    TAGS=$(yq -r '.. | select(.filters?) | .filters.tags[]' < "anta_catalog.yml" | sort -u)
+
+    parallel --tag './nrfu_tag.sh {} nrfu_reports' ::: $TAGS || NRFU_STATUS=$?
+
+    exit $NRFU_STATUS
+    ```
+
+    Change `anta_catalog.yml` to match your catalog file specified in `ANTA_CATALOG` of `nrfu-tag.sh`.
+
+3. Make both scripts executable:
+
+    ```bash
+    chmod +x nrfu-tag.sh run-tag.sh
+    ```
+
+4. Run the script:
+
+    ```bash
+    ./run-tag.sh
+    ```
+
+5. The script will execute ANTA tests for each tag in parallel, generating a JSON report for each tag in the `nrfu_reports` directory.
+
+### Performance Considerations
+
+#### Parallel Job Control
+
+The `-j` option in `GNU Parallel` controls the number of concurrent jobs. Consider these factors when setting the job limit:
+
+1. **Available CPU cores**:
+
+     - Recommend: jobs = number of cores - 1
+     - Example: For 8 cores, use `-j7`
+
+2. **Memory requirements**:
+
+     - Each ANTA instance requires memory
+     - Monitor system resources during execution using `htop` or similar tools
+
+#### JSON vs YAML
+
+When working with large catalogs and inventories, using JSON format instead of YAML can provide better performance as ANTA loads JSON files more efficiently. If you choose to use JSON:
+
+- Convert your YAML files to JSON (you can use `yq` for this)
+- Modify the scripts to use `jq` instead of `yq`:
+
+  ```bash
+  # For device-based implementation
+  NODE_NAMES=$(jq -r '.anta_inventory.hosts[].name' < "anta_inventory.json")
+
+  # For tag-based implementation
+  TAGS=$(jq -r '.. | select(.filters?) | .filters.tags[]' < "anta_catalog.json" | sort -u)
+  ```
+
+  Example of converting YAML to JSON:
+
+  ```bash
+  yq -o=json eval 'anta_inventory.yaml' > anta_inventory.json
+  yq -o=json eval 'anta_catalog.yaml' > anta_catalog.json
+  ```
+
+### Results Management
+
+ANTA can output results in various formats:
```

Comment:
do you need to repeat these? I would just point to the NRFU page documentatin and say you use JSON (less maintenance)